pkg/cmd/roachtest/test_runner.go

```diff
@@ -712,8 +719,14 @@ func (r *testRunner) runWorker(
 		if artifactsRootDir == "" {
 			artifactsRootDir, _ = os.MkdirTemp("", "roachtest-logger")
 		}
-
-		escapedTestName := teamCityNameEscape(testToRun.spec.Name)
+		testName := testToRun.spec.Name
+		// N.B. c may be nil owing to clusterCreateErr
+		if c != nil && c.arch != vm.ArchAMD64 {
+			// N.B. For non-default cpu architecture, encode it in the test name. This helps to differentiate test
+			// artifacts/results by cpu architecture.
+			testName = fmt.Sprintf("%s/cpu_arch=%s", testName, c.arch)
```

Comment:
I wonder if this will be confusing. AFAICT, this `testName` is only used when reporting the test name on TC. Everywhere else (github issue, source code, etc) will use the name without suffix.

Reply:
We have `ROACHTEST_arch=` parameter in GH. Short of serializing test specs as part of artifacts, I don't believe I found a less hackier way of encoding the architecture to ensure the artifacts (for amd64 and arm64) don't get mixed. Did you have another idea?